src/SafeDNS/TemplateClient.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace UKFast\SDK\SafeDNS;
+
+use UKFast\SDK\SafeDNS\Entities\Record;
+use UKFast\SDK\SafeDNS\Entities\Template;
+
+class TemplateClient extends Client
+{
+    const TEMPLATE_MAP = [
+        'created_at' => 'createdAt'
+    ];
+
+    const RECORD_MAP = [
+        'created_at' => 'createdAt'
+    ];
+
+    /**
+     * Returns a paginated list of available templates
+     * @param int $page
+     * @param int $perPage
+     * @param array $filters
+     * @return int|\UKFast\SDK\Page
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getPage($page = 1, $perPage = 15, $filters = [])
+    {
+        $page = $this->paginatedRequest('v1/templates', $page, $perPage);
+        $page->serializeWith(function ($item) {
+            return $this->serializeTemplate($item);
+        });
+
+        return $page;
+    }
+
+    /**
+     * Gets a specific template by id
+     * @param $id
+     * @return array
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getById($id)
+    {
+        $response = $this->get('v1/templates/' . $id);
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return $this->serializeTemplate($body->data);
+    }
+
+    /**
+     * Gets a paginated list of records for the template
+     * @param $id
+     * @param int $page
+     * @param int $perPage
+     * @param array $filters
+     * @return int|\UKFast\SDK\Page
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getRecords($id, $page = 1, $perPage = 15, $filters = [])
```

Comment:
My only question is should this be split into two separate clients, one for templates and one for template records? I believe we have this logical split in other places in the SDK

Reply:
I was a bit confused with regards to this as some people seem to but them in one client if the share similar paths for example `v1/templates` and `v1/templates/' . $id . '/records` then some split them.